concepts/observability-configurable-logging/08-revised-api.md

```diff
@@ -0,0 +1,290 @@
+-> do we need the LogPreset concept?
+
+### Support of additional inputs
+It turned out that there are typical inputs available which you would like to process as well, mainly the kubelet logs
+-> add a source selector
+
+## Revised API
+
+More and more oppinion/abstraction was added to the `LogPipeline` concept and the meaningfullnes of the `LogPreset` is in the room. We could now try to push down the `LogPipeline` to be more native and accepting that the user can do really stupid things in combination of supported plugins and revise the LogPreset concept.
+Or we push up even more the `LogPipeline` API by introducing the convinience of the LogPreset while keeping native support as a customization option clearly indicating an unsupported scenario.
+This proposal favors the second idea by dropping the LogPreset fully and having only one layer which is already an abstraction. Giving no chance to configure fluent-bit natively. For that you can simply go with a custom installation.
+
+```YAML
+kind: LogPipeline
+apiVersion: telemetry.kyma-project.io/v1alpha1
+metadata:
+  name: OpenSearchHTTP-App
+spec:
+  input:
+    application: # maps to central tail pipeline
+      namespaces: [] # maps to rewrite_tag rule
+      excludeNamespaces: []
+      pods: []
+      excludePods: []
+      containers: []
+      excludeContainers: []
+      matchLabels:
+        app: icke
+    system: {} #maps to systemd based input like kubelet logs
+    custom: | # entering unsupported mode
+      Name dummy
+      Dummy {"message":"dummy"}
+  filters:
+    - multilineParser:
+        type: "java"
+    - parser:
+        type: "json" # support of built-in parsers
+    - modify:
+        add:
+          key: cluster_identifier
+          value: ${KUBERNETES_SERVICE_HOST}
+    - modify:
+        remove:
+          key: cluster_identifier
+    - select:
+        include:
+          key:
+          regexp:
+        exclude:
+          key:
+          regexp:
+    - custom: | # no "Match" available, entering unsupported mode
+        Name    record_modifier
+        Record  cluster_identifier ${KUBERNETES_SERVICE_HOST}
+
+  output: #only one output
+    http:
+      Dedot: true
+      Host:
+        value: "icke.com"
+      HTTP_User:
+        value: "icke"
+        fromSecretKeyRef:
+          name: my-elastic-credentials
+          namespace: default
+          key: ES_ENDPOINT
+      HTTP_Password:
+        fromSecretKeyRef:
+          name: my-elastic-credentials
+          namespace: default
+          key: ES_ENDPOINT
+    custom:
+      | # no "Match" available, no filebuffer settings available, entering unsupported mode
+      Name               es
+      Alias              es-output
+      Host               ${ES_ENDPOINT} # Defined in Secret
+      HTTP_User          ${ES_USER} # Defined in Secret
+      HTTP_Password      ${ES_PASSWORD} # Defined in Secret
+      LabelMapPath       /files/labelmap.json
+
+  custom:
```

Comment:
Not sure if custom is a good name here. It might give the impression that everything here is unsupported. Should custom parsers be unsupported?

Reply:
If there is something supported, then it should not below that, I agree. About parsers... can a custom parser be harmful? Probably not.. but also goes back to the comment above about moving parsers into an own CRD. Introducing a Parsers CRD sounds very fluentbit specific and should cover other central configs in future.. maybe we can keep them for now in the pipeline itself and compile the central config out of it till we have a better understanding on what else will be central?

So maybe let's keep all as non-custom indeed. Defining parsers/multilineparsers will not be harmful at all as long as you don't use them, same for variables and files.

Reply:
custom on root level removed